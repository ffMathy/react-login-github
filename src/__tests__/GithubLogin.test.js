import React from 'react';
import LoginGithub from '../LoginGithub.js';
import PopupWindow from '../PopupWindow.js';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

test('Renders defaults', () => {
  const component = renderer.create(
    <LoginGithub clientId="foo" redirectUri="http://foo.test/auth/github"/>
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('Renders with `className`', () => {
  const component = renderer.create(
    <LoginGithub clientId="foo" redirectUri="http://foo.test/auth/github" className="foobar"/>
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('Renders with `buttonText`', () => {
  const component = renderer.create(
    <LoginGithub clientId="foo" redirectUri="http://foo.test/auth/github" buttonText="Foo"/>
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('Opens OAuth dialog', () => {
  const clientId = 'foo';
  const redirectUri = 'http://foo.test/auth/github';

  const component = (
    <LoginGithub clientId={clientId} redirectUri={redirectUri}/>
  );
  const wrapper = shallow(component);

  wrapper.find('button').simulate('click');

  const query = `client_id=${clientId}&scope=user:email&redirect_uri=${redirectUri}`

  expect(wrapper.instance().popup.url).toBe(
    `https://github.com/login/oauth/authorize?${query}`
  );
});
