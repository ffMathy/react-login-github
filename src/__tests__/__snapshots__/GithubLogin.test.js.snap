// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders defaults 1`] = `
<button
  className=""
  disabled={false}
  onClick={[Function]}
>
  Sign in with GitHub
</button>
`;

exports[`Renders with \`buttonText\` 1`] = `
<button
  className=""
  disabled={false}
  onClick={[Function]}
>
  Foo
</button>
`;

exports[`Renders with \`className\` 1`] = `
<button
  className="foobar"
  disabled={false}
  onClick={[Function]}
>
  Sign in with GitHub
</button>
`;
